letsencrypt/client/revoker.py

```diff
@@ -6,146 +14,513 @@
 
 import Crypto.PublicKey.RSA
 import M2Crypto
-import zope.component
 
 from letsencrypt.acme import messages
 from letsencrypt.acme import util as acme_util
 
-from letsencrypt.client import crypto_util
-from letsencrypt.client import display
-from letsencrypt.client import interfaces
+from letsencrypt.client import errors
+from letsencrypt.client import le_util
 from letsencrypt.client import network
 
+from letsencrypt.client.display import util as display_util
+from letsencrypt.client.display import revocation
+
 
 class Revoker(object):
     """A revocation class for LE.
 
+    .. todo:: Add a method to specify your own certificate for revocation - CLI
+
+    :ivar network: Network object
+    :type network: :class:`letsencrypt.client.network`
+
+    :ivar installer: Installer object
+    :type installer: :class:`~letsencrypt.client.interfaces.IInstaller`
+
     :ivar config: Configuration.
-    :type config: :class:`letsencrypt.client.interfaces.IConfig`
+    :type config: :class:`~letsencrypt.client.interfaces.IConfig`
 
-    """
+    :ivar bool no_confirm: Whether or not to ask for confirmation for revocation
 
-    def __init__(self, installer, config):
+    """
+    def __init__(self, installer, config, no_confirm=False):
         self.network = network.Network(config.server)
         self.installer = installer
         self.config = config
+        self.no_confirm = no_confirm
+
+        le_util.make_or_verify_dir(config.cert_key_backup, 0o700, os.geteuid())
 
-    def acme_revocation(self, cert):
-        """Handle ACME "revocation" phase.
+        # TODO: Find a better solution for this...
+        self.list_path = os.path.join(config.cert_key_backup, "LIST")
+        # Make sure that the file is available for use for rest of class
+        open(self.list_path, "a").close()
```

Comment:
I would remove this check:
1. you don't catch any exceptions, so the caller will just get generic (and thus cryptic) errors...
2. this call doesn't guard against any success failures, https://docs.python.org/2/glossary.html#term-eafp